src/components/Play/Play.js

```diff
@@ -66,37 +72,46 @@ const Play = ({ intl }) => {
     return () => clearTimeout(timeout);
   }, [pollingState]);
 
+  // Smart polling
   useEffect(() => {
     if (remaining) {
       joinTimeout && clearTimeout(joinTimeout);
-      // Smart polling
       setJoinTimeout(setTimeout(() => handleJoin(true), remaining));
     }
+    return () => clearTimeout(joinTimeout);
   }, [remaining]);
 
+  const playJSX = () => {
```

Comment:
I agree on putting all the JSX code in a separate method (really cool!), but what do you think about renaming this to `renderPlay` or something like that? Please, don't hate me.

_(This is also hapenning in the SearchSongs.js file)_

Reply:
Resolved in [this commit](https://github.com/rogue-sound/rogue-sound-web/pull/4/commits/3961e6938dd4473e95c22757ca881c0c59fc9992).